### PR TITLE
Add top-level testdata crate

### PR DESCRIPTION
**Pull request by sffc**
This adds a new crate "icu-testdata" with a function returning a data provider that loads from test data.  It includes a convenient command for re-generating the test data in a reproducible way.

### CODE REVIEW

**Comment by sffc** [comment]
This is ready for review.  Please let me know where you think this belongs in the directory heirarchy.

File: resources/testdata/src/bin/icu4x-gen-testdata.rs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    options.verbose = matches.is_present("VERBOSE");
+    let mut exporter = FilesystemExporter::try_new(json_serializer, &options)?;
+
+    for key in keys.iter() {
+        let result = provider.export_key(key, &mut exporter);
+        // Ensure flush() is called, even when the result is an error
+        exporter.flush()?;
+        result?;
+    }
+    */
```
Comment (sffc) [inline resources/testdata/src/bin/icu4x-gen-testdata.rs:194]: Reminder to self: delete these lines.

File: resources/testdata/src/bin/icu4x-gen-testdata.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+use icu_data_provider::iter::IterableDataProvider;
+use icu_fs_data_provider::export::fs_exporter;
+use icu_fs_data_provider::export::serializers;
+use icu_fs_data_provider::export::FilesystemExporter;
+use icu_fs_data_provider::manifest;
+use simple_logger::SimpleLogger;
+use std::fmt;
+use std::path::PathBuf;
+
+// #[derive(Debug)]
```
Comment (zbraniecki) [inline resources/testdata/src/bin/icu4x-gen-testdata.rs:14]: uncomment?

**Review by zbraniecki — DISMISSED** [review]
lgtm!

File: resources/testdata/src/bin/icu4x-gen-testdata.rs (hunk 2 above)
Comment (sffc) [reply resources/testdata/src/bin/icu4x-gen-testdata.rs:14]: Will remove.

**Review by zbraniecki — APPROVED** [review]
Lgtm!

**Pull request merged**